scapy/layers/dns.py

```diff
@@ -19,12 +19,161 @@
 from scapy.sendrecv import sr1
 from scapy.layers.inet import IP, DestIPField, UDP, TCP
 from scapy.layers.inet6 import DestIP6Field
-from scapy.error import warning
+from scapy.error import warning, Scapy_Exception
 from functools import reduce
 import scapy.modules.six as six
 from scapy.modules.six.moves import range
 
 
+def dns_get_str(s, p, pkt=None, _internal=False):
+    """This function decompresses a string s, from the character p.
+    params:
+     - s: the string to decompress
+     - p: start index of the string
+     - pkt: (optional) an InheritOriginDNSStrPacket packet
+
+    returns: (decoded_string, end_index, left_string)
+    """
+    # The _internal parameter is reserved for scapy. It indicates
+    # that the string provided is the full dns packet, and thus
+    # will be the same than pkt._orig_str. The "Cannot decompress"
+    # error will not be prompted if True.
+    max_length = len(s)
+    name = b""  # The result = the extracted name
+    burned = 0  # The "burned" data, used to determine the remaining bytes
+    q = None  # Will contain the index after the pointer, to be returned
+    processed_pointers = [p]  # Used to check for decompression loops
+    while True:
+        if abs(p) >= max_length:
+            warning("DNS RR prematured end (ofs=%i, len=%i)" % (p, len(s)))
+            break
+        cur = orb(s[p])  # current value of the string at p
+        p += 1  # p is now pointing to the value of the pointer
+        burned += 1
+        if cur & 0xc0:  # Label pointer
+            if q is None:
+                # p will follow the pointer, whereas q will not
+                q = p + 1
+            if p >= max_length:
+                warning("DNS incomplete jump token at (ofs=%i)" % p)
+                break
+            p = ((cur & ~0xc0) << 8) + orb(s[p]) - 12  # Follow the pointer
+            burned += 1
+            if pkt and hasattr(pkt, "_orig_s") and pkt._orig_s:
+                # There should not be a loop as pkt is None
+                name += dns_get_str(pkt._orig_s, p, None, _internal=True)[0]
+                if burned == max_length:
+                    break
+            elif p in processed_pointers:
+                warning("DNS decompression loop detected")
+                break
+            elif not _internal:
+                raise Scapy_Exception("DNS message can't be compressed at this point!")  # noqa: E501
```

Comment:
Could you rewrite this line to remove the noqa ?
For example:
msg = "DNS message can't be compressed at this point!"
raise Scapy_Exception(msg)